GVFS/GVFS.Service/GVFSService.Windows.cs

```diff
@@ -365,5 +370,40 @@ private DirectorySecurity GetServiceDirectorySecurity(string serviceDataRootPath
 
             return serviceDataRootSecurity;
         }
+
+        private void LaunchServiceUIIfNotRunning(int sessionId)
+        {
+            NamedPipeClient client;
+            using (client = new NamedPipeClient(GVFSConstants.Service.UIName))
+            {
+                if (!client.Connect())
+                {
+                    this.TerminateExistingProcess(GVFSConstants.Service.UIName);

```

Comment:
Help me understand why we are killing the process and restarting it if we can't connect?

Reply:
This is old code that is reused ([here is the original](https://github.com/microsoft/VFSForGit/blob/686b3b1f82021b515bf1d4dce728b9b90925a13a/GVFS/GVFS.Service/Handlers/NotificationHandler.Windows.cs#L19) in master.)

And probably is not needed. @wilbaker thoughts?

Reply:
> And probably is not needed. @wilbaker thoughts?

I'm not familiar with the service UI code.  Have you tried going back in history (before the move to GitHub) to see if you can find any discussion on why the code is structured this way?